src/ray/raylet/node_manager.cc

```diff
@@ -1866,11 +1866,10 @@ void NodeManager::HandleCancelResourceReserve(
   }
 
   // Return bundle resources.
-  if (ReturnBundleResources(bundle_spec)) {
-    // Call task dispatch to assign work to the released resources.
-    TryLocalInfeasibleTaskScheduling();
-    DispatchTasks(local_queues_.GetReadyTasksByClass());
-  }
+  ReturnBundleResources(bundle_spec);
```

Comment:
Why delete if judgment？

Reply:
This looks OK-- probably harmless to always retry and a little better in terms of defensive coding.

Reply:
ah, I just removed as an experiment (to see if it changes anything). I think you are right.